gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationMapperUtils.java

```diff
@@ -111,4 +114,31 @@ protected static RCMRMT030101UK04Authorise extractSupplyAuthorise(RCMRMT030101UK
             .findFirst()
             .orElse(null);
     }
+
+    public static Optional<DateTimeType> extractDispenseRequestPeriodStart(RCMRMT030101UK04Authorise supplyAuthorise) {
+        if (supplyAuthorise.hasEffectiveTime() && supplyAuthorise.getEffectiveTime().hasCenter()
+            && !supplyAuthorise.getEffectiveTime().getCenter().hasNullFlavor()) {
+            return Optional.of(DateFormatUtil.parseToDateTimeType(supplyAuthorise.getEffectiveTime().getCenter().getValue()));
+        }
+        if (supplyAuthorise.hasEffectiveTime() && supplyAuthorise.getEffectiveTime().hasLow()
+            && !supplyAuthorise.getEffectiveTime().getLow().hasNullFlavor()) {
+            return Optional.of(DateFormatUtil.parseToDateTimeType(supplyAuthorise.getEffectiveTime().getLow().getValue()));
+        }
+        if (supplyAuthorise.hasAvailabilityTime()) {
+            return Optional.of(DateFormatUtil.parseToDateTimeType(supplyAuthorise.getAvailabilityTime().getValue()));
+        }
+        return Optional.empty();
+    }
+
+    public static Stream<RCMRMT030101UK04MedicationStatement> extractAllMedications(RCMRMT030101UK04Component4 component4) {
+        return Stream.concat(
+            Stream.of(component4.getMedicationStatement()),
+            component4.hasCompoundStatement()
+                ? CompoundStatementUtil.extractResourcesFromCompound(component4.getCompoundStatement(),
+                    RCMRMT030101UK04Component02::hasMedicationStatement, RCMRMT030101UK04Component02::getMedicationStatement)
+                .stream()
+                .map(RCMRMT030101UK04MedicationStatement.class::cast)
+                : Stream.empty()
```

Comment:
Could you move this fragment to a separate method for readability?

Reply:
certainly